docs/ref/templates/builtins.txt

```diff
@@ -1838,18 +1838,21 @@ To escape each element of a sequence, use the :tfilter:`escapeseq` filter.
 ``escapejs``
 ------------
 
-Escapes characters for use in JavaScript strings. This does *not* make the
-string safe for use in HTML or JavaScript template literals, but does protect
-you from syntax errors when using templates to generate JavaScript/JSON.
+Escapes characters for use in JavaScript strings (text written inside single
+or double quotes). Using this filter only makes the string safe when used as
+the whole content of a JavaScript or JSON string literal (please see the
+example below). Note that it does not make the string safe for use in
+"JavaScript template literals" (i.e. the JavaScript backtick syntax), and
+any other use is not supported. It is generally recommended however that
+data should be passed using HTML ``data-`` attributes, or the
+:tfilter:`json_script` filter, rather than in embedded JavaScript.
```

Comment:
Maybe:
```suggestion
example below). It does not make the string safe for use in *"JavaScript
template literals"* (i.e. the JavaScript backtick syntax), or any other use. It
is generally recommended that data should be passed using HTML ``data-``
attributes, or the :tfilter:`json_script` filter, rather than in embedded
JavaScript.
```

Reply:
Seems ok to me, except that the sentence `It does not make the string safe for use in *"JavaScript
template literals"* (i.e. the JavaScript backtick syntax), or any other use.` is essentially a regression that re-introduces the exact problem this PR is intended to solve, i.e. that the documentation says that the filter is never safe to use, which is not true and extremely confusing. Removing the `or any other use` and instead putting `- any other use is not supported` after `Using this filter only makes the string safe when used as the whole content of a JavaScript or JSON string literal (please see the example below)` might be an improvement?

Reply:
I like this suggestion to phrase it positively by what's supported, rather than negatively by what's not. I think we can shorten to:

```
Using this filter only makes the string safe when used as the whole content of a JavaScript or JSON string literal, as below.
```

Reply:
Let's remove "i.e." - many English speakers don't understand what it means, and it's not necessary here.